src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -1054,33 +1052,473 @@ export const zlibInvalidInput = {
+      'ed ac sem sit amet arcu malesuada fermentum. Nunc sed. ';
+    const expectedBase64Deflate =
+      'eJxdUUtOQzEMvMoc4OndgT0gJCT2buJWlpI4jePeqZfpmX' +
+      'AKLRKbLOzx/HK73q6vOrhCunlF1qIDJhNUeW5I2ozT5OkD' +
+      'lKWLJWkncJG5403HQXAkT3Jw29B9uIEmToMukglZ0vS6oc' +
+      'iBh4JG8sV4oVLEUCitK2kxq1WzPnChHDzsaGKy491LofoA' +
+      'bWh8do43oeuYhB5EPCjcLjzYJo48KrfQBvnJecNFJvHT1+' +
+      'RSQsGoC7dn2t/xjhduTA1NWyQIZR0pbHwMDatnD+crPqKS' +
+      'qGPHp1vnlsWM/07ubf7bheF7kqSj84Bm0R1fYTfaK8vqqq' +
+      'fKBtNMhe3OZh6N95CTvMX5HJJi4xOVzCgUOIMSLH7wmeOH' +
+      'aFE4RdpnGavKtrB5xzfO/Ll9';
+    const expectedBase64Gzip =
+      'H4sIAAAAAAAAA11RS05DMQy8yhzg6d2BPSAkJPZu4laWkjiN4' +
+      '96pl+mZcAotEpss7PH8crverq86uEK6eUXWogMmE1R5bkjajN' +
+      'Pk6QOUpYslaSdwkbnjTcdBcCRPcnDb0H24gSZOgy6SCVnS9Lq' +
+      'hyIGHgkbyxXihUsRQKK0raTGrVbM+cKEcPOxoYrLj3Uuh+gBt' +
+      'aHx2jjeh65iEHkQ8KNwuPNgmjjwqt9AG+cl5w0Um8dPX5FJCw' +
+      'agLt2fa3/GOF25MDU1bJAhlHSlsfAwNq2cP5ys+opKoY8enW+' +
+      'eWxYz/Tu5t/tuF4XuSpKPzgGbRHV9hN9ory+qqp8oG00yF7c5' +
+      'mHo33kJO8xfkckmLjE5XMKBQ4gxIsfvCZ44doUThF2mcZq8q2' +
+      'sHnHNzRtagj5AQAA';
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.deflate(inputString, (err, buffer) => {
+        zlib.inflate(buffer, (err, inflated) => {
+          assert.strictEqual(inflated.toString(), inputString);
+          resolve();
+        });
+      });
+      await promise;
+    }
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.gzip(inputString, (err, buffer) => {
+        // Can't actually guarantee that we'll get exactly the same
+        // deflated bytes when we compress a string, since the header
+        // depends on stuff other than the input string itself.
+        // However, decrypting it should definitely yield the same
+        // result that we're expecting, and this should match what we get
+        // from inflating the known valid deflate data.
+        zlib.gunzip(buffer, (err, gunzipped) => {
+          assert.strictEqual(gunzipped.toString(), inputString);
+          resolve();
+        });
+      });
+      await promise;
+    }
+
+    let buffer = Buffer.from(expectedBase64Deflate, 'base64');
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.unzip(buffer, (err, buffer) => {
+        assert.strictEqual(buffer.toString(), inputString);
+        resolve();
+      });
+
+      await promise;
+    }
+
+    buffer = Buffer.from(expectedBase64Gzip, 'base64');
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.unzip(buffer, (err, buffer) => {
+        assert.strictEqual(buffer.toString(), inputString);
+        resolve();
+      });
+      await promise;
+    }
+  },
+};
+
+// Test taken from
+// https://github.com/nodejs/node/blob/fc02b88f89f8d5abf5ee4414a1026444c18d77b3/test/parallel/test-zlib-empty-buffer.js
+// TODO: enable brotli cases once implemented
```

Comment:
We use TODO(soon) to differentiate ourselves from Node.js' TODO comments:

```suggestion
// TODO(soon): enable brotli cases once implemented
```